src/main/java/racingcar/Application.java

```diff
@@ -1,7 +1,128 @@
+        List<Car> cars = createCars(carNames);
+        // 시도 횟수 입력
+        int tryCount = getTryCount();
+        // 레이싱 게임 실행 -> 한 라운드 결과 출력
+        runRacingGame(cars, tryCount);
+        // 우승자 목록 찾기 -> 우승자 목록 출력
+        printWinners(findWinners(cars));
+    }
+
+    // 자동차 이름 입력
+    private static List<String> getCarNames() {
+//        Scanner scanner = new Scanner(System.in);
+        List<String> carNames;
+        while (true) {
+            System.out.println("경주할 자동차 이름을 입력하세요. (이름은 쉼표(,) 기준으로 구분)");
+            String input = Console.readLine();
+            carNames = Arrays.asList(input.split(",", -1));
+            try { // 예외가 발생한다면? -> catch
+                validateCarNames(carNames); // 유효성 검사
+                break;
+            }
+            catch (IllegalArgumentException e) {
+                System.out.println("[ERROR] 올바른 형식의 자동차 이름을 입력하세요.");
+            }
+        }
+        return carNames;
+    }
+
+    // 자동차 이름 유효성 검사
+    private static void validateCarNames(List<String> carNames) {
+        for (String str : carNames) {
+            if (str.isEmpty() || str.length() > 5) {
+                throw new IllegalArgumentException();
+            }
+        }
+        Set<String> uniqueNames = new HashSet<>(carNames); // 중복을 허용하지 않는 자료구조
+        if (uniqueNames.size() < carNames.size()) { // 작다면 중복된 이름이 있다는 것을 의미함
+            throw new IllegalArgumentException();
+        }
+    }
+
+    // 시도 횟수 입력
+    private static int getTryCount() {
+        int tryCount;
+        while (true) {
+            System.out.println("시도할 횟수는 몇 회인가요?");
+            String input = Console.readLine();
+            try {
+                tryCount = Integer.parseInt(input); // 문자열을 정수로 변환하는 문법
+                if (tryCount <= 0) {
+                    throw new IllegalArgumentException();
+                }
+                break;
+            }
+            catch (IllegalArgumentException e) {
+                System.out.println("[ERROR] 올바른 형식의 숫자를 입력하세요.");
+            }
+        }
+        return tryCount;
+    }
+
+    // 자동차 목록 생성.
+    private static List<Car> createCars(List<String> carNames) {
+        List<Car> cars = new ArrayList<>(); // 리스트 생성 후 리스트의 원소가 Car 크랫스의 인스턴스가 되도록
+        for (String name : carNames) {
+            cars.add(new Car(name));
+        }
+        return cars;
+    }
+
+    // 레이싱 게임 실행
+    private static void runRacingGame(List<Car> cars, int tryCount) {
+        System.out.println("\n실행 결과");
+        for (int i = 0; i < tryCount; i++) {
+            for (Car car : cars) {
+                car.moveForward();
+            }
+            printRoundResult(cars);
+        }
+    }
+
+    // 한 라운드 결과 출력
+    private static void printRoundResult(List<Car> cars) {
+        for (Car car : cars) {
+//            System.out.println(car.getName() + " : " + "-".repeat(car.getPosition()));
+            System.out.println(car.getName() + " : " + new String(new char[car.getPosition()]).replace('\0', '-'));
```

Comment:
```suggestion
            System.out.println(car.name + " : " + new String(new char[car.getPosition()]).replace('\0', '-'));
```
Car 클래스의 name 필드 접근 제어자를 private에서 public 으로 변경하면 다음과 같이 코드를 수정할 수 있습니다.
그런데 문제에는 이런 요구사항이 있었죠.
`name, position 변수의 접근 제어자인 private을 변경할 수 없다.`

왜 필드를 private으로 사용해야할까요?

Reply:
멤버 변수로의 직접 접근은 좋지 않습니다. 객체 지향 프로그래밍의 원칙 중 하나인 캡슐화에 위배됩니다.
캡슐화는 외부에서의 직접 접근을 막고, 객체에 접근하는 메서드를 통해서만 접근이 가능하게 합니다. 정보를 숨겨 안정성을 높이고 유지보수, 재사용에 용의하게 합니다.

Reply:
왜 외부에서 직접 접근하면 안될까요?

Reply:
외부에서 직접 접근하는 메서드가 많아진다면 추후 클래스의 구성이나 구현을 변경할 때 수정이 어려워지기 때문입니다!